src/Pact/Eval.hs

```diff
@@ -481,6 +484,25 @@ evaluateDefs info defs = do
 
       return (d', dn, mapMaybe (either Just (const Nothing)) $ toList d')
 
+    resolveBareModRef f fn memo
+        | fn /= moduleBareName mdef = resolveModRef f (ModuleName fn Nothing)
+        | otherwise = liftIO (readIORef memo) >>= \case
+            Just cachedMR -> return $ Just cachedMR
+            Nothing -> do
+              mdef' <- (_MDModule . mInterfaces . traverse) (resolveIfs f) mdef
+              let mr = Just $ Ref $ mkModRef f mdef'
+              liftIO $ writeIORef memo mr
+              pure mr
+
+    resolveIfs i mn = do
+      resolveModule (getInfo i) mn >>= \case
+        Nothing -> evalError info $ "Modref Interface not defined: " <> pretty mn
+        Just (ModuleData (MDInterface Interface{..}) _irefs) -> pure _interfaceName
```

Comment:
Assuming this is the most common branch we should hit, maybe it should be first?

Reply:
not necessarily most common. you can argue that you want to fail faster from a gas perspective